Dockerfile

```diff
@@ -1,40 +1,50 @@
-FROM osrf/ros:noetic-desktop
+FROM ros:noetic-ros-core
 
 # Use bash instead of sh
 SHELL ["/bin/bash", "-c"]
 
 # Update Ubuntu Software repository
 RUN apt update && \
-    apt install -y \
+    apt install --no-install-recommends -y \
         git \
+        build-essential \
         python3-dev \
-        python3-pip
-
-RUN pip3 install \
-        conan \
+        python3-pip && \
+    pip3 install \
+        conan==1.59 \
         rosdep \
         vcstool && \
     conan config set general.revisions_enabled=1 && \
     conan profile new default --detect > /dev/null && \
-    conan profile update settings.compiler.libcxx=libstdc++11 default 
+    conan profile update settings.compiler.libcxx=libstdc++11 default && \
+    rm -rf /var/lib/apt/lists/*
```

Comment:
We don't need to remove apt lists at this stage since it's gonna be removed after packaged build process

Reply:
It is not necessary but should slightly reduce step size

Reply:
That's fine